src/openms_gui/source/VISUAL/TOPPASScene.cpp

```diff
@@ -480,26 +488,64 @@ namespace OpenMS
 
   bool TOPPASScene::isEdgeAllowed_(TOPPASVertex* u, TOPPASVertex* v)
   {
-    if (u == 0 ||
-        v == 0 ||
-        u == v ||
+    if (u == 0 || v == 0 || u == v ||
         // edges leading to input files make no sense:
         qobject_cast<TOPPASInputFileListVertex*>(v) ||
         // neither do edges coming from output files:
         qobject_cast<TOPPASOutputFileListVertex*>(u) ||
-        // nor edges from input to output without a tool in between:
-        (qobject_cast<TOPPASInputFileListVertex*>(u)
-        && qobject_cast<TOPPASOutputFileListVertex*>(v)) ||
-        // nor multiple incoming edges for a single output file/list node
-        (qobject_cast<TOPPASOutputFileListVertex*>(v)
-        && v->inEdgesBegin() != v->inEdgesEnd()) ||
-        // nor mergers connected directly to an output node
-        (qobject_cast<TOPPASMergerVertex*>(u)
-        && qobject_cast<TOPPASOutputFileListVertex*>(v)))
+        // nor edges from input/merger/splitter directly to output:
+        ((qobject_cast<TOPPASInputFileListVertex*>(u) || qobject_cast<TOPPASMergerVertex*>(u) || qobject_cast<TOPPASSplitterVertex*>(u)) && qobject_cast<TOPPASOutputFileListVertex*>(v)) ||
+        // nor multiple incoming edges for an output or splitter node:
```

Comment:
A splitter should allow multiple input edges. The code changes should be easy.
-- 
Sent via mobile phone.

Am 15. September 2015 10:40:53 MESZ, schrieb hendrikweisser <notifications@github.com>:
>>          // edges leading to input files make no sense:
>>          qobject_cast<TOPPASInputFileListVertex*>(v) ||
>>          // neither do edges coming from output files:
>>          qobject_cast<TOPPASOutputFileListVertex*>(u) ||
>> -        // nor edges from input to output without a tool in between:
>> -        (qobject_cast<TOPPASInputFileListVertex*>(u)
>> -        && qobject_cast<TOPPASOutputFileListVertex*>(v)) ||
>> -        // nor multiple incoming edges for a single output file/list
>node
>> -        (qobject_cast<TOPPASOutputFileListVertex*>(v)
>> -        && v->inEdgesBegin() != v->inEdgesEnd()) ||
>> -        // nor mergers connected directly to an output node
>> -        (qobject_cast<TOPPASMergerVertex*>(u)
>> -        && qobject_cast<TOPPASOutputFileListVertex*>(v)))
>> +        // nor edges from input/merger/splitter directly to output:
>> +        ((qobject_cast<TOPPASInputFileListVertex*>(u) ||
>qobject_cast<TOPPASMergerVertex*>(u) ||
>qobject_cast<TOPPASSplitterVertex*>(u)) &&
>qobject_cast<TOPPASOutputFileListVertex*>(v)) ||
>> +        // nor multiple incoming edges for an output or splitter
>node:
>
>I'm not sure what you're referring to. What do you think should be
>changed?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/OpenMS/OpenMS/pull/1593/files#r39486591